src/Mod/Part/App/TopoShape.cpp

```diff
@@ -2412,16 +2412,14 @@ TopoDS_Shape TopoShape::makeLongHelix(Standard_Real pitch, Standard_Real height,
     Handle(Geom_Surface) surf;
     Standard_Boolean isCylinder;
 
-    if (angle < Precision::Confusion()) {                                      // Cylindrical helix
+    if (angle < Precision::Confusion() && angle > -Precision::Confusion()) {                                      // Cylindrical helix

```

Comment:
This change allows makeLongHelix support negative angles. It works fine, and does not hurt any legacy functionality as I can see.